src/random_room_placement.rs

```diff
@@ -0,0 +1,278 @@
+use std::fmt;
+use rand::distributions::WeightedIndex;
+use rand::prelude::*;
+use rand::rngs::StdRng;
+use rand::Rng;
+use serde::Serialize;
+use crate::error::Result;
+
+struct RandomRoomLevel {
+    level: Level,
+}
+
+byond_fn!(fn random_room_generate(width, height, desired_room_count, hash) {
+    random_room_gen(width, height, desired_room_count, hash).ok()
+});
+
+fn random_room_gen(width_as_str: &str,
+    height_as_str: &str,
+    desired_room_count_as_str: &str,
+    hash_as_str: &str,
+    )
+    -> Result<String>{
+    let default_hash: u64 = rand::thread_rng().gen();
+    let width = width_as_str.parse::<i32>()?;
+    let height = height_as_str.parse::<i32>()?;
+    let desired_room_count = desired_room_count_as_str.parse::<i32>()?;
+
+    //let seed: &str = Alphanumeric.sample_string(&mut rand::thread_rng(), 32).as_str();
+
+    let mut rng: StdRng = SeedableRng::seed_from_u64(hash_as_str.parse::<usize>()?.try_into().unwrap_or(default_hash));
+
+
+    let level = RandomRoomLevel::new(width, height, desired_room_count, &mut rng);
+
+    Ok(serde_json::to_string(&level.all_rooms)?)
+}
+
+impl RandomRoomLevel {
+    fn new(
+        width: i32,
+        height: i32,
+        desired_room_count: i32,
+        rng: &mut StdRng,
+
+    ) -> Level {
+        let level = Level::new(width, height);
+
+        let mut map = RandomRoomLevel { level };
+
+        map.place_rooms_random(desired_room_count, rng);
+        map.level
+    }
+
+    fn place_rooms_random(&mut self, desired_room_count: i32, rng: &mut StdRng) {
```

Comment:
Tbh is it worth just making place_rooms_random a function of `Level` instead of creating an entirely different `RandomRoomLevel` struct?